src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp

```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright (C) 2008-2019 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "ScriptedGossip.h"
+#include "MotionMaster.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "TemporarySummon.h"
+
+enum ThalorienDawnseekerTexts
+{
+    SAY_THALORIEN_1 = 0,
+    SAY_THALORIEN_2 = 1,
+    SAY_THALORIEN_3 = 2,
+    SAY_THALORIEN_4 = 3,
+    SAY_THALORIEN_5 = 4,
+    SAY_THALORIEN_6 = 5,
+    SAY_THALORIEN_7 = 6,
+    SAY_THALORIEN_8 = 7,
+    SAY_THALORIEN_9 = 8,
+    SAY_THALORIEN_10 = 9,
+
+    SAY_MORLEN_1 = 0,
+    SAY_MORLEN_2 = 1,
+    SAY_MORLEN_3 = 2,
+    SAY_MORLEN_4 = 3,
+    SAY_MORLEN_5 = 4
+};
+
+enum ThalorienDawnseekerSummons
+{
+    SUMMON_SUNWELL_DEFENDER = 1,
+    SUMMON_SCOURGE_ZOMBIE   = 2,
+    SUMMON_GHOUL_INVADER    = 3,
+    SUMMON_CRYPT_RAIDER     = 4
+};
+
+enum ThalorienDawnseekerActions
+{
+    ACTION_START_QUEST = 1
+};
+
+enum ThalorienDawnseekerEvents
+{
+    EVENT_INTRO_0           = 1,
+    EVENT_INTRO_1           = 2,
+    EVENT_INTRO_2           = 3,
+    EVENT_INTRO_3           = 4,
+    EVENT_SALUTE            = 5,
+    EVENT_DEFENDERS_RUN     = 6,
+    EVENT_DEFENDERS_RUN_2   = 7,
+    EVENT_THALORIEN_GO      = 8,
+    EVENT_INTRO_4           = 9,
+    EVENT_SUMMON_MORLEN     = 10,
+    EVENT_INTRO_5           = 11,
+    EVENT_MORLEN_1          = 12,
+    EVENT_SPAWN_WAVE_1      = 13,
+    EVENT_SPAWN_WAVE_2      = 14,
+    EVENT_SPAWN_WAVE_3      = 15,
+    EVENT_WAVE_ATTACK       = 16,
+    EVENT_MORLEN_2          = 17,
+    EVENT_MORLEN_ATTACK     = 18,
+    EVENT_OUTRO_1           = 19,
+    EVENT_OUTRO_2           = 20,
+    EVENT_OUTRO_3           = 21,
+    EVENT_OUTRO_4           = 22,
+    EVENT_KNEEL             = 23
+};
+
+enum ThalorienDawnseeker
+{
+    MENU_EXAMINE_REMAINS        = 10908,
+
+    NPC_THALORIEN_DAWNSEEKER    = 37205,
+    NPC_SUNWELL_DEFENDER        = 37211,
+    NPC_MORLEN_GOLDGRIP         = 37542,
+    NPC_SCOURGE_ZOMBIE          = 37538,
+    NPC_GHOUL_INVADER           = 37539,
+    NPC_CRYPT_RAIDER            = 37541,
+
+    SPELL_BLOOD_PRESENCE        = 50689,
+    SPELL_KILL_CREDIT           = 70265,
+    SPELL_POLYMORPH_VISUAL      = 27123
+};
+
+float const defenderReverse = 5.67232f;
+Position const defenderRun[10] =
+        {
```

Comment:
identation and dont use magic numbers for array size
constexpr uint32 DEFENDER_RUN_SIZE { 10 };